### ISSUE TITLE
Validation error in generated XSD catalog for DTD files

### ISSUE THREAD

**Issue by WolfgangHG** [eclipse/wildwebdeveloper#847]
I installed the "JBoss Tools" plugin, which adds the JavaEE schemas to the old Eclipse XML catalog, and this one is converted to "system-catalog.xml".
When opening this file and the setting "Enable schema based validation" is "Always"...
![schemabasedvalidation](https://user-images.githubusercontent.com/138049/184533395-941feda1-4453-4aea-88a9-ba56ca503e5e.png)

... there are errors reported for all DTD files:
![validationerror](https://user-images.githubusercontent.com/138049/184533401-2823a4c1-c342-422e-bfbd-60a3c4fec2d1.png)

`cvc-pattern-valid: Value '-//Sun Microsystems, Inc.//DTD J2EE Application 1.3//EN' is not facet-valid with respect to pattern '[a-zA-Z0-9\-'\(\)+,./:=?;!*#@$_%]*' for type 'publicIdentifier'.`

It seems the space is missing in the RegEx for CDATA? The attribute "publicIdentifier" is defined as "CDATA" according to "catalog.dtd".

According to "application_1_3.dtd", this is valid:

```
<!--
This is the XML DTD for the J2EE 1.3 application deployment
descriptor.  All J2EE 1.3 application deployment descriptors
must include a DOCTYPE of the following form:

  <!DOCTYPE application PUBLIC
	"-//Sun Microsystems, Inc.//DTD J2EE Application 1.3//EN"
	"http://java.sun.com/dtd/application_1_3.dtd">

-->


```

**Comment by angelozerr** [comment]
Coud you share please your catalog.xml?

**Comment by WolfgangHG** [comment]
Sorry, I forgot it. Here it is.
[system-catalog.zip](https://github.com/eclipse/wildwebdeveloper/files/9339332/system-catalog.zip)

**Comment by mickaelistria** [comment]
Can you reproduce this issue with latest release or snapshots?

**Comment by WolfgangHG** [comment]
Yes - I downloaded "eclipseide-wild-web-developer-1.0.3-win32.win32.x86_64.zip", then installed JBoss Tools according to https://tools.jboss.org/downloads/jbosstools/2022-12/4.26.0.Final.html (I installed only three plugins in group "* Abridged JBoss Tools": "JBoss AS, WildFly & EAP Server Tools", "JBoss Runtime Detection Core" and "JBoss Tools Java Standard Tools").
When opening the catalog, the same error ist shown.
